### PR TITLE
Get player refactor

### PR DESCRIPTION
**Pull request by bticknor**
PlayerService.getPlayer now checks to see if the ID passed is a valid slack user id (prefixed by UD), if so returning a Some(Player) built with that id, and if not returning None

### CODE REVIEW

File: src/main/scala/elo/PlayerService.scala [hunk 1]
```diff
@@ -4,17 +4,24 @@ import com.redis.serialization.Parse.Implicits.parseDouble
...
-      // If a user doesn't have a score in the DB, assign them the default
-      val userScore = redisClient.get[Double](slackUserId).getOrElse(EloRankingSystem.initialScore)
-      Player(slackUserId, userScore)
+      None
     }
-  }
+}
 
-  def findAllPlayers: Seq[Player] = {
+  def findAllPlayers: Seq[Option[Player]] = {
```
[the remark this answers is not in the document]
Comment (bticknor) [reply src/main/scala/elo/PlayerService.scala]: Ah I see

File: src/main/scala/elo/Bot.scala [hunk 2]
```diff
@@ -124,28 +119,30 @@ object PingPongBot extends App {
         slackClient.sendMessage(message.channel, leaderboard)
       }
 
-      // get ID of first other user mentioned
-      val challengee = mentionedIds
-        .find(_ != selfId) // retrieves first element for which the find condition is true
-        .map(PlayerService.playerService.getPlayer)
-        .getOrElse(Player.Nobody)
+      // this will never throw an exception since messages necessarily have a user
+      val challenger = PlayerService.playerService.getPlayer(message.user).get
```
[the remark this answers is not in the document]
Comment (bticknor) [reply src/main/scala/elo/Bot.scala]: > Calling .get on an option is a code smell in general - typically it's only acceptable in non-production code.

I had a feeling this was the case, was curious to see what you'd say about it

[the remark this answers is not in the document]
Comment (bticknor) [reply src/main/scala/elo/Bot.scala]: > I imagine there are more functional programming-friendly ways to write the message processing logic, but we'll cross that bridge when we get there 😉

Was thinking the same thing

File: src/main/scala/elo/Bot.scala [hunk 3]
```diff
@@ -124,28 +120,31 @@ object PingPongBot extends App {
...
+        .find(_ != selfId) // option[string]
+        .getOrElse("nobody")
+      val challengeeOpt = PlayerService.playerService.getPlayer(challengeeID)
 
       // if its a challenge, send a challenge message
       if(message.text.contains("hallenge")) {
-        val challenger = PlayerService.playerService.getPlayer(message.user)
-        val chalMessage = challengeMessage(challenger, challengee)
+        // need valid identities for both players for a challenge
+        val chalMessage = (challengerOpt, challengeeOpt) match {
```
Comment (bticknor) [inline src/main/scala/elo/Bot.scala]: @wongkk25 what about this...having both challenger and challengee as options and pattern matching that we have both of them in order to generate a challenge / report a loss

Comment (wongkk25) [inline src/main/scala/elo/Bot.scala]: You could also write this in a for comprehension, as such:
```scala
val chalMessage = (for {
  challenger <- challengerOpt
  challengee <- challengeeOpt
} yield {
  challengeMessage(challenger, challengee)
}).getOrElse("Mention a user to challenge them!")

Comment (wongkk25) [reply src/main/scala/elo/Bot.scala]: Oops just saw this comment, my response is here https://github.com/bticknor/elo-ranking-slackbot/pull/10#discussion_r257334864

**Pull request merged**